browser/ui/views/bookmarks/brave_bookmark_context_menu_unittest.cc

```diff
@@ -116,3 +118,21 @@ TEST_F(BraveBookmarkContextMenuTest, ShowAllBookmarksButtonMenuCheckedState) {
   EXPECT_TRUE(bookmark_context_menu->IsItemChecked(
       IDC_TOGGLE_ALL_BOOKMARKS_BUTTON_VISIBILITY));
 }
+
+TEST_F(BraveBookmarkContextMenuTest,
+       ShowAllBookmarksButtonMenuVisibilityState) {
+  auto bookmark_context_menu = CreateBookmarkContextMenu();
+  EXPECT_FALSE(bookmark_context_menu->IsCommandVisible(
+      IDC_TOGGLE_ALL_BOOKMARKS_BUTTON_VISIBILITY));
+
+  std::u16string title = u"xyz";
+  GURL url("https://www.xyz.com");
+  const BookmarkNode* node =
```

Comment:
I was wondering that the pointer node can cause memory leak as it is not getting freed. But then this is a test case. Does it matter to free a pointer pointed memory in test case too? 
I'm not sure but possibly Remove method must be already doing that!🤔

Reply:
As TreeNode owns its child nodes, we don't need to free them, I think.

```
//   using MyModel = ui::TreeNodeModel<MyNode>;
template <class NodeType>
class TreeNode : public TreeModelNode {
 public:
  using TreeNodes = std::vector<std::unique_ptr<NodeType>>;
```